File: vital-configs/src/main/java/me/xra1ny/vital/configs/VitalConfigsSubModule.java
package me.xra1ny.vital.configs;

import me.xra1ny.essentia.inject.annotation.Component;
import me.xra1ny.vital.VitalSubModule;
import org.jetbrains.annotations.NotNull;

/**
 * Defines the vital-configs submodule.
 */
@Component
public class VitalConfigsSubModule extends VitalSubModule {
    @Override
    public @NotNull String getName() {
        return "vital-configs";
    }
}

File: vital-configs/src/main/kotlin/me/vitalframework/configs/VitalConfigsSubModule.kt
package me.vitalframework.configs

import me.vitalframework.SubModule
import me.vitalframework.VitalCoreSubModule.Companion.logger
import me.vitalframework.VitalSubModule
import org.springframework.stereotype.Component
import org.yaml.snakeyaml.constructor.Constructor

typealias SnakeYamlConstructor = Constructor

/**
 * Defines the official vital-configs submodule, which is displayed when Vital starts.
 * It contains the Vital configs system, which can be used to create class-based configs.
 */
@SubModule("vital-configs")
class VitalConfigsSubModule(
    val vitalConfigs: List<VitalConfig>,
) : VitalSubModule() {
    val logger = logger()

    override fun onInstall() {
        for (vitalConfig in vitalConfigs) {
            logger.info("Config '${vitalConfig::class.java.name}' successfully registered")
        }
    }
}
